### ISSUE TITLE
pack error on Windows 10 17746

### ISSUE THREAD

**Issue by WangZishi** [rustwasm/wasm-pack#277]
## 🐛 Bug description
I just try the hello world on my computer.
Project structure is

![default](https://user-images.githubusercontent.com/8288105/44888820-a20cf500-ad05-11e8-89f5-ade2f151f66a.png)

`wasm-pack init` is success
![default](https://user-images.githubusercontent.com/8288105/44888939-3b3c0b80-ad06-11e8-80ab-0884f589602b.png)

but `wasm-pack pack . ` is faild
![default](https://user-images.githubusercontent.com/8288105/44888755-56f2e200-ad05-11e8-8100-a7576764615a.png)

wasm-pack.log
```
Aug 31 10:21:31.981 INFO Running pack command...
Aug 31 10:21:31.982 INFO Path: Some(".")
Aug 31 10:21:31.982 INFO Packing up the npm package...
Aug 31 10:21:31.982 ERRO 系统找不到指定的文件。 (os error 2)
```

#### 🤔 Expected Behavior
No error occurred

#### 👟 Steps to reproduce
1. create new hello world project
2. wasm-pack init
3. wasm-pack pack .

#### 🌍 Your environment
os: windows 10 x64 17746.1000
wasm-pack version: 0.4.2
rustc version: 1.30.0-nightly (02cb8f2a4 2018-08-29)

**Comment by fitzgen** [comment]
Thanks for the bug report!

**Comment by csmoe** [comment]
could you provide your local npm meta info? I suspect that `npm` wasn't installed on your machine.

**Comment by csmoe** [comment]
@WangZishi I suspect that `npm` wasn't installed on your machine, could you paste the `npm` meta info here?

------
 
@fitzgen 
anyway, `pack` cmd invokes `npm` directly https://github.com/rustwasm/wasm-pack/blob/89be40df2f478a19b9fdd38dafdc7792967560f2/src/npm.rs#L10-L18
maybe better to add a binary exsit_or_not checking as you did in `test` PR here?

**Comment by fitzgen** [comment]
That sounds good to me! We should probably always do that, just to get good error messages to users

**Comment by WangZishi** [comment]
@csmoe 
![default](https://user-images.githubusercontent.com/8288105/45197818-d0ef1200-b295-11e8-8ff1-bd70af8d786d.png)

**Comment by ashleygwilliams** [comment]
ah @WangZishi what happens when you run `wasm-pack pack` (without the dot?). the dot shouldn't cause an issue but it may be...

**Comment by ashleygwilliams** [comment]
hrm. thanks so much for the response and sorry this is taking so long to figure out. @steveklabnik do you think you could try this locally?

**Comment by steveklabnik** [comment]
I get

```console
C:\Users\steve> wasm-pack pack
Unable to find the pkg directory at path ".", or in a child directory of "."
```

**Comment by csmoe** [comment]
this bug can be reproduced with Chinese-ver Windows7 in my virtualbox, comparing to stevelabnik's comment, I suspect that system locale is the one to be blamed.

**Comment by luotaoyeah** [comment]
this might help: https://github.com/rust-lang/rust/issues/42791

**Comment by danwilhelm** [comment]
Unfortunately, this issue affects login, pack, and publish on Windows!

I verified that the cause suggested by @luotaoyeah is indeed occurring. `Command` places `"npm"` in quotes when calling it, which causes it to fail in Windows only.

The [recommended solution is to call `npm` using `cmd /c`](https://github.com/rust-lang/rust/pull/44542). I am working on a PR for this and have verified the `cmd /c` solution works. Unfortunately, it requires some OS-specific code -- but it seems unavoidable.

**Comment by ashleygwilliams** [comment]
@danwilhelm are you on discord or IRC? feel free to reach out to me on either platfrom (i'm ag_dubs)

**Comment by ashleygwilliams** [comment]
closed by #489! about to go out!

**Comment by danwilhelm** [comment]
@WangZishi @csmoe I was not able to verify this issue was resolved on Chinese-ver Windows! Could you possibly verify whether it is fixed in [wasm-pack 0.6.0](https://crates.io/crates/wasm-pack/0.6.0)?

**Comment by csmoe** [comment]
@danwilhelm thanks for the fix. I'll test then(my locale gussing maybe almost not the cause).
